### DISCUSSION TITLE
Tutorial install on CPanel shared hosting (no docker)

### DISCUSSION THREAD

**Discussion by ewwink** [ptrumpis/snap-camera-server#80] [Share]
look the file `.env`, uncomment and set the following

```
# values will be set by Docker if you run docker compose
DB_HOST=localhost
DB_PORT=3306
DB_NAME=dbname
DB_USER=dbuser
DB_PASS=dbpass
#PORT=5645
STORAGE_SERVER=https://mysubdomain.yoururl.com/snapcamera
STORAGE_PATH=/home/cpuser/mysubdomain.yoururl.com/snapcamera
IMPORT_DIR=import
MEDIA_DIR=snap-camera-media
MEDIA_DIR_ALT=snap-camera-media-alt
```

Create directories with the above values in root of web app, check `stderr.log` if there missing directory or error.

in `Snap Camera.exe` replace `studio-app.snapchat.com` (23 chars) with `mysubdomain.yoururl.com` (23 chars)  with hex editor.

web based hex patcher needed 😁 or use manual web hex editor `https://hexed.it` 

For NodeJS 14 and below you  need to change function `.replaceAll('oldvalue', 'newvalue'), 'gi')` with `.replace(/oldvalue/gi, 'newvalue')`

CPanel need loader file, instead using `server.js` as **Application startup file** create  `loadapp.cjs` or any name with `.cjs` extension with the following content

```
async function loadapp() {
    await import("/home/cpuser/mysubdomain.yoururl.com/server.js");
}

loadapp();

```

done

**Comment by ptrumpis** [comment]
Thank you for sharing these instructions!

**Comment by ewwink** [comment]
you're welcome, and thank you too for this awesome app, can you change 3 `replaceAll()` to `replace()` in https://github.com/ptrumpis/snap-camera-server/blob/ab62900c6fadfa7a091792c718e683f433d7ef92/src/utils/helper.js#L187

**Reply by ptrumpis** [reply]
Sure, I will change them.
You can also provide a pull request if you want to.
